keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
 type ResourceClaimSpec struct {
-    // ResourceClassName references the driver and additional parameters
-    // via the name of a ResourceClass that was created as part of the
-    // driver deployment.
+    // This field holds configuration for multiple potential drivers which
+    // could satisfy requests in this claim. The configuration applies to
+    // the entire claim. It is ignored while allocating the claim.
+    //
     // +optional
-    ResourceClassName string
+    // +listType=atomic
+    Config []ConfigurationParameters
 
-    // ParametersRef references a separate object with arbitrary parameters
-    // that will be used by the driver when allocating a resource for the
-    // claim.
+    // These constraints must be satisfied by the set of devices that get
+    // allocated for the claim.
+    Constraints []Constraint
+
+    // Requests are individual requests for separate resources for the claim.
+    // An empty list is valid and means that the claim can always be allocated
+    // without needing anything. A class can be referenced to use the default
+    // requests from that class.
     //
-    // The object must be in the same namespace as the ResourceClaim.
-    // +optional
-    ParametersRef *ResourceClaimParametersReference
-}
-```
+    // +listType=atomic
+    Requests []Request
 
-The `ResourceClassName` field may be left empty. The parameters are sufficient
-to determine which driver needs to provide resources. This leads to some corner cases:
-- Empty `ResourceClassName` and nil `ParametersRef`: this is a claim which requests
-  no resources. Such a claim can always be allocated with an empty result. Allowing
-  this simplifies code generators which dynamically fill in the resource requests
-  because they are allowed to generate an empty claim.
-- Non-empty `ResourceClassName`, nil `ParametersRef`, nil
-  `ResourceClass.DefaultClaimParametersRef`: this is handled the same way, the
-  only difference is that the cluster admin has decided that such claims need
-  no resources by not providing default parameters.
-
-There is no default ResourceClass. If that is desirable, then it can be
-implemented with a mutating and/or admission webhook.
+    // Future extension, ignored by older schedulers. This is fine because scoring
+    // allows users to define a preference, without making it a hard requirement.
+    //
+    //
+    // Score *SomeScoringStruct
 
-```
-// ResourceClaimStatus tracks whether the resource has been allocated and what
-// the resulting attributes are.
-type ResourceClaimStatus struct {
-    // Allocation is set by the resource driver once a resource or set of
-    // resources has been allocated successfully. If this is not specified, the
-    // resources have not been allocated yet.
+    // Shareable indicates whether the allocated claim is meant to be shareable
+    // by multiple consumers at the same time.
     // +optional
-    Allocation *AllocationResult
+    Shareable bool
```

Comment:
The default is `true`, correct?

Reply:
No, it's false. We can turn this into a pointer and say "default if unset is true". I'm not a fan of defaulting, but I guess for "kubectl get" it would make sense to make that default explicit.

Or would it make sense to remove this field entirely and just silently mark all allocated claims as "sharable"?

Reply:
I think "sharable" or "not applicable" (meaning users just don't reference the claim in more than one place) is the common case, and "exclusive" is the uncommon case.

Does it make sense to reverse this then and call it "Exclusive"?

Although, you might be right -- it might make sense to just remove this entirely. I think it made sense as a class field in the past, but maybe not as much as a claim field, especially now since classes are optional.

Reply:
We can go further and also remove it from the allocation result. If we don't have a good use case for it, we shouldn't implement the logic in the scheduler either.

Agreed?

Reply:
I am happy to drop it.

Reply:
You can ignore this. I'm going to remove it.